jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/partial/RowBased.java

```diff
@@ -15,6 +15,7 @@
  */
 package com.salesforce.datacloud.jdbc.core.partial;
 
+import com.salesforce.datacloud.jdbc.core.DataCloudStatement;
```

Comment:
layering-wise, this dependency feels a bit backwards. Not sure waht `RowBased` does exactly, but it feels like this should live a layer below `DataCloudStatement`?